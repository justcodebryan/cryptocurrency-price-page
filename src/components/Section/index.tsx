import classNames from 'classnames'
import { DetailedHTMLProps, HTMLAttributes, PropsWithChildren } from 'react'

import styles from './styles.module.scss'

type SectionProps = PropsWithChildren<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>>

const Section = ({ style, className, children }: SectionProps) => {
  const internalClass = classNames([styles['section'], className])

  return (
    <div className={internalClass} style={style}>
      {children}
    </div>
  )
}

export default Section
