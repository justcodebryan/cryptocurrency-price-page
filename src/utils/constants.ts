import { NoopFn } from '@/types'

export const NOOP: NoopFn = () => void 0

export const DEFAULT_MODAL_CONTAINER_ID = 'modal-container'

export const DEFAULT_POLLING_INTERVAL = 10 * 1000
