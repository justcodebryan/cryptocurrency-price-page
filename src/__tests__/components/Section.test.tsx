import Section from '@/components/Section'
import { toJSON } from '@/utils/test-utils'
import renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react'

describe('Section Test Suit', () => {
  test('Section Snapshot', () => {
    const component = renderer.create(<Section />)
    const tree = toJSON(component)
    expect(tree).toMatchSnapshot()
  })

  test('Should show children element', () => {
    const { container } = render(<Section>test</Section>)
    expect(container.querySelector('div')).toBeDefined()
    expect(screen.getByText('test')).toBeDefined()
  })
})
