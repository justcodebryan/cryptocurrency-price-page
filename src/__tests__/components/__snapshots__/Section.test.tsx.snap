// Vitest Snapshot v1

exports[`Section Test Suit > Section Snapshot 1`] = `
<div
  className="_section_8f1029"
/>
`;
